### PR TITLE
Improving eventhub closing connection logic

### PR DESCRIPTION
**Pull request by HiranyaKavishani**
### Purpose
This Pr will improve the logic for detect connection failures in event receivers

### Issues

Fixes #

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/messaging/connection.go [hunk 1]
```diff
@@ -44,14 +44,35 @@ func connectToRabbitMQ(url string) (*amqp.Connection, error) {
...
+	}
+
+	if shouldReconnect {
 		c.conn.Close()
 		c, rabbitConn, err = connectionRetry(key)
 		if err != nil {
 			logger.LoggerMsg.Errorf("Cannot establish connection for topic %s", key)
 		}
+	} else {
+		logger.LoggerMsg.Infof("NotifyClose from the connection and channel are %v and %v respectively, NotifyBlocked from the connection is %v",
```
Comment (Rajith90) [inline adapter/internal/messaging/connection.go:74]: The order of variables seems incorrect with the error messag

File: adapter/internal/messaging/km_listener.go [hunk 2]
```diff
@@ -48,11 +48,12 @@ func handleKMConfiguration(deliveries <-chan amqp.Delivery, done chan error) {
...
 		// var keyManagerConfig resourceTypes.KeymanagerConfig
 		var kmConfigMap map[string]interface{}
-
-		// var eventType string
-		json.Unmarshal([]byte(string(d.Body)), &notification)
-
-
+		unmarshalErr := json.Unmarshal([]byte(string(d.Body)), &notification)
+		if unmarshalErr != nil {
+			logger.LoggerMsg.Errorf("Error occured while unmarshalling event data %v", unmarshalErr)
```
Comment (Rajith90) [inline adapter/internal/messaging/km_listener.go:53]:
```suggestion
			logger.LoggerMsg.Errorf("Error occured while unmarshalling key manager event data %v", unmarshalErr)
```

File: adapter/internal/messaging/km_listener.go [hunk 3]
```diff
@@ -62,11 +63,13 @@ func handleKMConfiguration(deliveries <-chan amqp.Delivery, done chan error) {
...
 
 		var decodedByte, err = base64.StdEncoding.DecodeString(notification.Event.PayloadData.Value)
+
 		if err != nil {
 			if _, ok := err.(base64.CorruptInputError); ok {
-				panic("\nbase64 input is corrupt, check the provided key")
+				logger.LoggerMsg.Error("\nbase64 input is corrupt, check the provided key")
 			}
-			panic(err)
+			logger.LoggerMsg.Errorf("Error occured %v", err)
```
Comment (Rajith90) [inline adapter/internal/messaging/km_listener.go:71]: Can this error message be more descriptive

File: adapter/internal/messaging/notification_listener.go [hunk 4]
```diff
@@ -72,18 +72,22 @@ func handleNotification(deliveries <-chan amqp.Delivery, done chan error) {
...
+		}
+		logger.LoggerMsg.Infof("Event %s is received", notification.Event.PayloadData.EventType)
 		var decodedByte, err = base64.StdEncoding.DecodeString(notification.Event.PayloadData.Event)
 		if err != nil {
 			if _, ok := err.(base64.CorruptInputError); ok {
-				panic("\nbase64 input is corrupt, check the provided key")
+				logger.LoggerMsg.Error("\nbase64 input is corrupt, check the provided key")
 			}
-			panic(err)
+			logger.LoggerMsg.Errorf("Error occured %v", err)
```
Comment (Rajith90) [inline adapter/internal/messaging/notification_listener.go:86]: Can this error message be more descriptive

File: adapter/internal/messaging/revoked_token_listener.go [hunk 5]
```diff
@@ -32,7 +32,12 @@ func handleTokenRevocation(deliveries <-chan amqp.Delivery, done chan error) {
 
 	for d := range deliveries {
 		var notification EventTokenRevocationNotification
-		json.Unmarshal([]byte(string(d.Body)), &notification)
+		unmarshalErr := json.Unmarshal([]byte(string(d.Body)), &notification)
+		if unmarshalErr != nil {
+			logger.LoggerMsg.Errorf("Error occured while unmarshalling event data %v", unmarshalErr)
```
Comment (Rajith90) [inline adapter/internal/messaging/revoked_token_listener.go:37]:
```suggestion
			logger.LoggerMsg.Errorf("Error occurred while un marshalling revoked token event data %v", unmarshalErr)
```

**Comment by HiranyaKavishani** [comment]
Changes are added with https://github.com/wso2/product-microgateway/pull/2001/files

**Pull request closed without merging**